source/model/backtraceMetrics.ts

```diff
@@ -157,7 +157,9 @@ export class BacktraceMetrics {
     const clientAttributes = this.attributeProvider() as {
       [index: string]: any;
     };
-    const result: { [index: string]: string } = {};
+    const result: { [index: string]: string } = {
+      'application.session': this.sessionId || ''
```

Comment:
the sessionId must exist here. We shouldn't even execute this code if the sessionId is unavailable. I think we should set it in the constructor and avoid conditions of using '' if it's empty.

Reply:
We need sessionId to be able to be undefined, since we get it's initial value from localStorage and we determine a "session launch" to be if there is no sessionId every set.

I replaced this code with `sessionId as string` to make it more clear that this is really just a type cast. The sessionId will be defined, or in any use case, should be defined. I could check explicitly in the function, but that seems like adding too much complexity and constraints on the `getAttributes` function, it should only get the attributes.

Reply:
Yeah but we read it in constructor. So after leaving the constructor we should ALWAYS have a sessionId, right?

Reply:
I refactored some code into the constructor so sessionId will never be undefined!